File: lib/tmdb_api/model/media_type.dart
class MediaType {
  final String name;

  const MediaType(this.name);

  static const all = MediaType("all");
  static const movie = MediaType("movie");
  static const tv = MediaType("tv");
  static const person = MediaType("person");

  static const values = [all, movie, tv, person];

  @override
  String toString() {
    return name;
  }

  factory MediaType.valueOf(String json) {
    if (json == null) return null;
    return values.firstWhere((v) => json == v.name);
  }
}

File: kmp/model/src/main/java/com/tikalk/tmdb/json/model/MediaType.kt
package com.tikalk.tmdb.json.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class MediaType {
    @SerialName("all")
    all,

    @SerialName("movie")
    movie,

    @SerialName("tv")
    tv,

    @SerialName("person")
    person
}
